### PR TITLE
Remove OldTastyInspector from scaladoc

### PR DESCRIPTION
**Pull request by nicolasstucki**
This a follow-up to https://github.com/lampepfl/dotty/pull/17623 after reverting in #17744.

[test_windows_full]

### CODE REVIEW

**Comment by nicolasstucki** [comment]
@Florian3k do you know what could be going wrong with that failing test?

**Comment by Florian3k** [comment]
From a quick look at this error and source code - I don't have any ideas.
I can try digging more into that later.

**Comment by Florian3k** [comment]
Unfortunately, I wasn't able to figure it out, and I suppose I won't get much further without Windows machine.

**Comment by sjrd** [comment]
Case-insensitive file systems ... 🤦‍♂️ 
https://github.com/lampepfl/dotty/pull/17751/files#diff-6a38cc2514e83acf57f7ad76e824efe6083fb569b45fd99386707ad7c4f19246

**Comment by sjrd** [comment]
The new failure on Windows is caused by #17229. See #17959 for the PR that should fix it.

File: scaladoc/test/dotty/tools/scaladoc/ExternalLocationProviderIntegrationTest.scala [hunk 1]
```diff
@@ -57,7 +57,8 @@ class Scaladoc3ExternalLocationProviderIntegrationTest extends ExternalLocationP
 
 def getScalaLibraryPath: String = {
   val classpath: List[String] = System.getProperty("java.class.path").split(java.io.File.pathSeparatorChar).toList
-  classpath.find(_.contains("scala-library-2")).getOrElse("foobarbazz") // If we don't find the scala 2 library, the test will fail
+  val stdlib = classpath.find(_.contains("scala-library-2")).getOrElse("foobarbazz") // If we don't find the scala 2 library, the test will fail
+  new java.io.File(stdlib).getCanonicalPath() // canonicalize for case-insensitive file systems
```
Comment (nicolasstucki) [inline scaladoc/test/dotty/tools/scaladoc/ExternalLocationProviderIntegrationTest.scala:61]: @sjrd are these the only changes that you did? 

This looks good.

**Comment by nicolasstucki** [comment]
@Florian3k could you double check that the diff of this PR matches the one that you reviewed (modulo the Windows fix)?

**Review by Florian3k — APPROVED** [review]
Yup, looks good.

**Pull request merged**